src/main/java/com/github/cameltooling/lsp/internal/parser/ParserFileHelperFactory.java

```diff
@@ -145,15 +146,25 @@ private boolean isCamelKYamlDSL(TextDocumentItem textDocumentItem, String uri) {
 		//improve this method to provide better heuristic to detect if it is a Camel file or not
 		return uri.endsWith(CAMELK_YAML_FILENAME_SUFFIX)
 				|| isYamlFileWithCamelKShebang(textDocumentItem, uri)
-				|| isYamlFileWithCamelKModelineLike(textDocumentItem, uri);
+				|| isYamlFileWithCamelKModelineLike(textDocumentItem, uri)
+				|| isYamlFileOfCRDType(textDocumentItem, uri);
+	}
```

Comment:
When conditional chains get too large and are in normal form (all conjunctions or all disjunctions), 
I prefer to put all the conditions inside a list of lambdas and then do a AllMatch or AnyMatch.

The list is much less error prone and easier to extend.

Ill put tomorrow an example commit to show what I mean

Reply:
I prefer the notation with only the || from a readability point of view. The code is shorter and have less parentheses.